### PR TITLE
Antag (Hud) Datums.

### LINKED ISSUE #10801: Wizard Apprentices don't get wiz hud elements
(WEB REPORT BY: gutlesscoward REMOTE: 172.93.109.202:7777)

> Revision 9709e5d58d69e0e097496e3af7870ca97eb44066
> 
> General description of the issue
> Apprentices made using the apprentice scroll can't see the little wizard hat icon that denotes someone as a wizard. this leads to problems when one is trying to stealth and the other is murdering everything on sight.
> 
> What you expected to happen
> I expected my apprentice to know that I was a wizard despite the disguise
> 
> What actually happened
> I got murdered to death by my apprentice, who didn't know I was a wizard.
> 
> Steps to reproduce if possible
> 1. Be a wizard
> 2. Buy apprentice scrolls. I picked muscle wizard.
> 3. remove your robes and wear a disguise.
> 4. meet up with your apprentice

### LINKED ISSUE #11368: Reverted Vampthrall still able to see his and other peoples V signs
Got enthralled and dethralled. After a while the gray V sign showed up again so I ahelped it.
According to Admins I wasn't listed as antag. Despite that I was still able to see mine and other peoples V sign. Might be exploitable.

Length of time in which bug has been known to occur:
No clue, was my first time as thrall.

Issue bingo:
- [ ] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [ ] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

### PR DESCRIPTION
**Pull request by ghost**
(Hud) Datums.
- [x] Antag Huds now use planes instead of really odd copypasted code
- [x] Antag Huds are datumised.
- [x] todo: see antags for admins command thing
- [x] todo: test antag huds
- [ ] todo: sechud & medhud. (maybe?)
- [ ] todo: test sechud & medhud
- [x] todo: personal antag huds; greytide implant, vampires, necromancer things, & thralls.
- [x] todo: test personal antag huds
- [x] adds a dunce hat for wizard apprentices

 (taken from the perspective of the mime)
bug:
if it is ticked, it has since been fixed.
- [x] doesn't work
- [x] doesn't work
- [x] doesn't work (I spent at least four hours on trying to just make it work)
- [x] rev icons don't work
- [x] if you right click people, the antag item is visible in the icon. (!!!!!)  http://www.byond.com/forum/?post=2128512 - will use image loc or whatever for this instead.-
- [x] toggling invisibility doesn't 100% work yet
- [x] everything is broken again
- [x] everyone is invisible
- [x] personal antag huds don't work
- [x] they now work too well; to the point where everyone can see them

100% tested:
- [x] Fixes #11368
- [x] Fixes #10801 
- [x] You cannot right click people to see if they are an antag.

### CODE REVIEW

**Comment by gbasood** [comment]
Since #11339 basically copy+pastes the antag hud code from before this, one of the two of these should be decidedly merged before the other

**Comment by ghost** [comment]
@gbasood make it yours, this is
a) highly adaptable
b) not working at all right now
c) not finished

File: code/_data_huds/_data_huds.dm [hunk 1]
```diff
@@ -0,0 +1,122 @@
+//
+
+#define CAN_SEE			1
```
Comment (ComicIronic) [inline code/_data_huds/_data_huds.dm:3]: Did you redefine TRUE and FALSE?

Comment (ghost) [inline code/_data_huds/_data_huds.dm:3]: what did he mean by this

**Comment by ghost** [comment]
I'll be taking a two-week break. This PR will be finished after that.

File: code/_data_huds/_data_huds.dm (hunk 1 above)
Comment (ComicIronic) [inline code/_data_huds/_data_huds.dm:3]: Did you create new defines for a bool, when we have TRUE and FALSE already?

Comment (ghost) [inline code/_data_huds/_data_huds.dm:3]: yeah because I'm :sunglasses:

**Comment by ghost** [comment]
Why did you close this @clusterfack

**Comment by clusterfack** [comment]
Because you've been consistently making system prs, and then not fixing the bugs with them, and then leaving for a few weeks after which you make another.

**Comment by Exxion** [comment]
Plus you said you'd finish it in two weeks two weeks ago
If you're still going to finish it, go ahead and someone'll probably reopen it if you ask
Or you can just open a new one

**Comment by ghost** [comment]
@Exxion I will

**Pull request closed without merging**